pkg/storage/tsdb/config.go

```diff
@@ -104,10 +106,11 @@ func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
 	f.StringVar(&cfg.Dir, "experimental.tsdb.dir", "tsdb", "directory to place all TSDB's into")
 	f.Var(&cfg.BlockRanges, "experimental.tsdb.block-ranges-period", "comma separated list of TSDB block ranges in time.Duration format")
 	f.DurationVar(&cfg.Retention, "experimental.tsdb.retention-period", 6*time.Hour, "TSDB block retention")
-	f.DurationVar(&cfg.ShipInterval, "experimental.tsdb.ship-interval", 1*time.Minute, "How frequently the TSDB blocks are scanned and new ones are shipped to the storage. 0 means shipping is disabled.")
-	f.IntVar(&cfg.ShipConcurrency, "experimental.tsdb.ship-concurrency", 10, "Maximum number of tenants concurrently shipping blocks to the storage.")
 	f.StringVar(&cfg.Backend, "experimental.tsdb.backend", "s3", "TSDB storage backend to use")
 	f.IntVar(&cfg.MaxTSDBOpeningConcurrencyOnStartup, "experimental.tsdb.max-tsdb-opening-concurrency-on-startup", 10, "limit the number of concurrently opening TSDB's on startup")
+	f.DurationVar(&cfg.CompactionInterval, "experimental.tsdb.compaction-interval", 2*time.Hour, "How frequently should locally-stored TSDB blocks be compacted (and shipped)")
```

Comment:
This is misleading. The blocks are compacted every 2hrs irrespective of this setting. I'd set it to something like 10mins by default. I know that most of the checks will be no-ops, but I'd rather check it more frequently than less frequently. It is likely that the blocks will become `5hrs` long with this setting (3hrs + 2hrs).

This is because when you call `db.Compact()`, it'll compact only if the `maxt - mint > 1.5 * blockRanges[0]`, else it'll shortcut and I think the default `blockRanges[0]` is `2h` anyways.

Reply:
> This is misleading. The blocks are compacted every 2hrs irrespective of this setting. I'd set it to something like 10mins by default. I know that most of the checks will be no-ops, but I'd rather check it more frequently than less frequently. It is likely that the blocks will become `5hrs` long with this setting (3hrs + 2hrs).
> 
> This is because when you call `db.Compact()`, it'll compact only if the `maxt - mint > 1.5 * blockRanges[0]`, else it'll shortcut and I think the default `blockRanges[0]` is `2h` anyways.

You're absolutely right... I was just looking at this, while adding metrics, and realized that 2 hours doesn't make sense. Prometheus itself tries to run this every minute (!) I've now set default to 5 mins. I should update the argument help too. I feel like this is such a low-level setting, that exposing it to user may cause more harm than good.

Reply:
Updated the help too.